CONTRIBUTING.md

```diff
@@ -96,6 +98,74 @@ Compile the JavaScript:
 ```shell
 ( cd ./app/frontend ; npm install ; npm run build )
 ```
+## Adding Hugging Face Models
+
+Follow the steps below to add a new Hugging Face model to your project:
+
+### 1. Create a New Template (Optional)
+- This step is optional and existing templates can be used.
+- Navigate to the `app/backend/templates` directory.
+- Create a new template following the format of the existing templates.
+- For a separate LLM, there should be **four separate files**:
+  - **`ask.prompty`**: Responsible for generating messages for the Ask approach.
+  - **`chat.prompty`**: Responsible for generating messages for the Chat approach.
+  - **`query.prompty`**: Used to generate search queries for AI search.
+  - **`tools.json`**: Contains custom functions for the LLM's Inference API calls. If not needed, this file can be empty.
```

Comment:
But how do I actually construct these? How do I know whether I even need a new template? What do I look for? Perhaps we could have some hints/insights here, like "go to the respective HF page, look for this thing and the other thing, create the template in this way and the other". It doesn't have to be the absolute truth, cause there is experimentation involved, but you can simply describe how **you** go about adding new templates (in a few words, nothing crazy).